factory/state/stateComponents_test.go

```diff
@@ -3,76 +3,152 @@ package state_test
 	componentsMock "github.com/multiversx/mx-chain-go/testscommon/components"
+	"github.com/multiversx/mx-chain-go/testscommon/factory"
 	"github.com/stretchr/testify/require"
 )
 
-func TestNewStateComponentsFactory_NilShardCoordinatorShouldErr(t *testing.T) {
+func TestNewStateComponentsFactory(t *testing.T) {
 	t.Parallel()
 
-	coreComponents := componentsMock.GetCoreComponents()
-	shardCoordinator := mock.NewMultiShardsCoordinatorMock(2)
-	args := componentsMock.GetStateFactoryArgs(coreComponents, shardCoordinator)
-	args.ShardCoordinator = nil
-
-	scf, err := stateComp.NewStateComponentsFactory(args)
-	require.Nil(t, scf)
-	require.Equal(t, errors.ErrNilShardCoordinator, err)
-}
-
-func TestNewStateComponentsFactory_NilCoreComponents(t *testing.T) {
-	t.Parallel()
-
-	coreComponents := componentsMock.GetCoreComponents()
-	shardCoordinator := mock.NewMultiShardsCoordinatorMock(2)
-	args := componentsMock.GetStateFactoryArgs(coreComponents, shardCoordinator)
-	args.Core = nil
-
-	scf, err := stateComp.NewStateComponentsFactory(args)
-	require.Nil(t, scf)
-	require.Equal(t, errors.ErrNilCoreComponents, err)
+	t.Run("nil Core should error", func(t *testing.T) {
+		t.Parallel()
+
+		coreComponents := componentsMock.GetCoreComponents()
+		args := componentsMock.GetStateFactoryArgs(coreComponents)
+		args.Core = nil
+
+		scf, err := stateComp.NewStateComponentsFactory(args)
+		require.Nil(t, scf)
+		require.Equal(t, errors.ErrNilCoreComponents, err)
+	})
+	t.Run("nil StatusCore should error", func(t *testing.T) {
+		t.Parallel()
+
+		coreComponents := componentsMock.GetCoreComponents()
+		args := componentsMock.GetStateFactoryArgs(coreComponents)
+		args.StatusCore = nil
+
+		scf, err := stateComp.NewStateComponentsFactory(args)
+		require.Nil(t, scf)
+		require.Equal(t, errors.ErrNilStatusCoreComponents, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		coreComponents := componentsMock.GetCoreComponents()
+		args := componentsMock.GetStateFactoryArgs(coreComponents)
+
+		scf, err := stateComp.NewStateComponentsFactory(args)
+		require.NoError(t, err)
+		require.NotNil(t, scf)
+	})
 }
 
-func TestNewStateComponentsFactory_ShouldWork(t *testing.T) {
+func TestStateComponentsFactory_Create(t *testing.T) {
 	t.Parallel()
 
-	coreComponents := componentsMock.GetCoreComponents()
-	shardCoordinator := mock.NewMultiShardsCoordinatorMock(2)
-	args := componentsMock.GetStateFactoryArgs(coreComponents, shardCoordinator)
-
-	scf, err := stateComp.NewStateComponentsFactory(args)
-	require.NoError(t, err)
-	require.NotNil(t, scf)
+	t.Run("CreateTriesComponentsForShardId fails should error", func(t *testing.T) {
+		t.Parallel()
+
+		coreComponents := componentsMock.GetCoreComponents()
+		args := componentsMock.GetStateFactoryArgs(coreComponents)
+		coreCompStub := factory.NewCoreComponentsHolderStubFromRealComponent(args.Core)
+		coreCompStub.InternalMarshalizerCalled = func() marshal.Marshalizer {
+			return nil
```

Comment:
so this will make CreateTriesComponentsForShardId fail?

Reply:
yes CreateTriesComponentsForShardId passes the marshaller on NewTrieFactory which will eventually fail